src/components/App/App.jsx

```diff
@@ -3,22 +3,66 @@ import Backdrop from '../Backdrop/Backdrop';
 import ModalSS from '../ModalSS/ModalSS';
 import ModalGal from '../ModalGal/ModalGal';
 import Hero from '../Hero/Hero';
+import dummyData from '../../../dummyData';
 
 class App extends Component {
   constructor(props) {
     super(props);
     this.state = {
-      testImgSrc: 'https://s3-us-west-1.amazonaws.com/airbnbusy/photos/p00020.jpg',
+      showModalGallery: false,
+      showModalShareSave: false,
     };
+    this.modalGalleryHandler = this.modalGalleryHandler.bind(this);
+    this.modalShareSaveHandler = this.modalShareSaveHandler.bind(this);
+    this.keyPressHandler = this.keyPressHandler.bind(this);
+  }
+
+  modalGalleryHandler() {
+    this.setState(prevState => ({ showModalGallery: !prevState.showModalGallery }));
+  }
+
+  modalShareSaveHandler(e) {
+    e.stopPropagation();
+    this.setState(prevState => ({ showModalShareSave: !prevState.showModalShareSave }));
+  }
+
+  // need to get this working
+  // might be tabIndex
+  // will probably set handler on modals
+  // onKeyDown={props.keyPressHandler}
+  // keyPressHandler={e => this.keyPressHandler(e)}
+
+  keyPressHandler(e) {
+    console.log('key pressed');
+    if (e.keyCode === 27) {
+      this.setState({
+        showModalGallery: false,
+        showModalShareSave: false,
+      });
+    }
   }
 
   render() {
+    // {console.log('dummyData', dummyData)}
+    // {console.log('heroImg', dummyData[0].uri)}
     return (
       <div>
-        <Backdrop />
+        <Backdrop
+          showModalGallery={this.state.showModalGallery}
+          showModalShareSave={this.state.showModalShareSave}
+          modalShareSaveHandler={e => this.modalShareSaveHandler(e)}
```

Comment:
Here, couldn't you just pass `this. modalShareSaveHandler `?

Reply:
Here I could not.  That prop gets passed to Button components which invoke the method on click.   These Button components are children of the Hero component.  Both the Hero and Buttons have unique click handlers.   I needed to take advantage of the synthetic event object's stopPropagation method to prevent event bubbling which would cause both the Hero and Button events to trigger when the button was clicked.

Reply:
I stand corrected.  I removed the anonymous function and event arg, and the app still exhibited the same behavior.  I was still able to reference the event object in my function definition.  Thanks!